src/nile/commands/compile.py

```diff
@@ -18,17 +18,22 @@ def compile_command(contracts):
         print(f"📁 Creating {ABIS_DIRECTORY} to store compilation artifacts")
         os.makedirs(ABIS_DIRECTORY, exist_ok=True)
 
+    all_contracts = contracts
+
     if len(contracts) == 0:
         print(f"🤖 Compiling all Cairo contracts in the {CONTRACTS_DIRECTORY} directory")
-        for path in get_all_contracts():
-            _compile_contract(path)
-    elif len(contracts) == 1:
-        _compile_contract(contracts[0])
-    else:
-        for path in contracts:
-            _compile_contract(path)
+        all_contracts = get_all_contracts()
 
-    print("✅ Done")
+    results = [_compile_contract(contract) for contract in all_contracts]
+    failures = sum(r != 0 for r in results)
+
+    if failures == 0:
+        print("✅ Done")
+    else:
+        exp = f"{failures} contract"
```

Comment:
is there an easy way to get _which ones_?

Reply:
Yeah, I thought about that as well. It's easy to get it done. What do you think about an output like this?

```
$  nile compile
🤖 Compiling all Cairo contracts in the contracts directory
🔨 Compiling contracts/app.cairo
🔨 Compiling contracts/contract2.cairo
contracts/contract2.cairo:22:5: Expected exactly 0 expressions, got 1.
    return (2)
    ^********^
🔨 Compiling contracts/contract.cairo
contracts/contract.cairo:19:5: Expected exactly 0 expressions, got 1.
    return (0)
    ^********^
🛑 Failed to compile the following 2 contracts:
   contracts/contract2.cairo
   contracts/contract.cairo
```

It's, of course, easy to adjust, so if you have any input, I can make it happen ;)

Reply:
that looks perfect!

Reply:
Done in the latest commit f04563d